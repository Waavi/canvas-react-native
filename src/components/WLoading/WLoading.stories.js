/* eslint-disable react-native/no-inline-styles */
import React from 'react'
import { storiesOf } from '@storybook/react-native'
import { boolean } from '@storybook/addon-knobs'
import { StoryView, StoryCase, StoryHeader } from '@modules/react-native-storybook/components'
import { Text } from 'react-native'
import { Colors } from '@/theme'
import { WLoading } from './WLoading'

storiesOf('WLoading', module)
	.add('Default', () => (
		<StoryView scrollable>
			<StoryHeader.h1 text="Loading" />
			<StoryCase text="with active = true" description="Use the Addons options">
				<WLoading active={boolean('Active', true)} />
			</StoryCase>
		</StoryView>
	))
	.add('colors', () => {
		const colors = [undefined, 'black']
		const bgColors = [undefined, 'white']
		return (
			<StoryView scrollable>
				<StoryHeader.h1 text="Loading colors" />
				{colors.map(color => (
					<StoryCase
						key={color || 'none'}
						text={color ? `with color '${color}'` : 'default'}
					>
						<WLoading active color={Colors[color]} />
					</StoryCase>
				))}
				<StoryHeader.h1 text="Loading background colors" />
				{bgColors.map(color => (
					<StoryCase
						key={color || 'none'}
						text={color ? `with background color '${color}'` : 'default'}
					>
						<WLoading active bgColor={Colors[color]} />
					</StoryCase>
				))}
			</StoryView>
		)
	})
	.add('position', () => {
		const textStyle = { margin: 20, textAlign: 'center' }
		return (
			<StoryView scrollable>
				<StoryHeader.h1 text="Loading" />
				<StoryCase text="default">
					<Text style={textStyle}>A sample text</Text>
					<WLoading active />
				</StoryCase>
				<StoryCase text="with position = absolute">
					<Text style={textStyle}>A sample text</Text>
					<WLoading active position="absolute" />
				</StoryCase>
			</StoryView>
		)
	})
	.add('opacity', () => {
		const textStyle = { margin: 20, textAlign: 'center' }
		return (
			<StoryView scrollable>
				<StoryHeader.h1 text="Loading" />
				<StoryCase text="default">
					<Text style={textStyle}>A sample text</Text>
					<WLoading active position="absolute" />
				</StoryCase>
				<StoryCase text="with opacity = 0.95">
					<Text style={textStyle}>A sample text</Text>
					<WLoading active position="absolute" opacity={0.95} />
				</StoryCase>
				<StoryCase text="with opacity = 0">
					<Text style={textStyle}>A sample text</Text>
					<WLoading active position="absolute" opacity={0} />
				</StoryCase>
			</StoryView>
		)
	})
