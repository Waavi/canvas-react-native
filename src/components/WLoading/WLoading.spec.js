import React from 'react'
import { shallow } from 'enzyme'
import { ActivityIndicator } from 'react-native'
import { WLoading } from './WLoading'

describe('Components/WLoading', () => {
	it('should be render ActivityIndicator native', () => {
		const wrapper = shallow(<WLoading active />)
		const innerIndicator = wrapper.find(ActivityIndicator)
		expect(innerIndicator).toBeTruthy()
		expect(wrapper).toMatchSnapshot()
	})
	it('should be not render ActivityIndicator native', () => {
		const wrapper = shallow(<WLoading active={false} />)
		const innerIndicator = wrapper.find(ActivityIndicator)
		expect(innerIndicator).toEqual({})
		expect(innerIndicator).toMatchSnapshot()
	})
})
