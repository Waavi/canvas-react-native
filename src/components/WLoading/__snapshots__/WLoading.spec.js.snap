// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components/WLoading should be not render ActivityIndicator native 1`] = `null`;

exports[`Components/WLoading should be render ActivityIndicator native 1`] = `
<_class
  cls={
    Array [
      "col-center",
      Object {
        "pos-absolute-fill": false,
      },
    ]
  }
>
  <_class
    cls="pos-absolute-fill"
    style={
      Object {
        "backgroundColor": "#EDF0F4",
        "opacity": 0.8,
      }
    }
  />
  <ActivityIndicator
    animating={true}
    color="#18E2A5"
    hidesWhenStopped={true}
    size="large"
  />
</_class>
`;
